pumpkin/src/entity/item.rs

```diff
@@ -0,0 +1,304 @@
+use crate::entity::player::Player;
+use crate::entity::{random_float, Entity};
+use crate::server::Server;
+use crate::world::World;
+use crossbeam::atomic::AtomicCell;
+use pumpkin_core::math::position::WorldPosition;
+use pumpkin_core::math::vector2::Vector2;
+use pumpkin_core::math::vector3::Vector3;
+use pumpkin_entity::entity_type::EntityType;
+use pumpkin_entity::pose::EntityPose;
+use pumpkin_entity::EntityId;
+use pumpkin_inventory::Container;
+use pumpkin_protocol::client::play::{CPickupItem, CSetEntityMetadata, CSpawnEntity, Metadata};
+use pumpkin_protocol::slot::Slot;
+use pumpkin_world::item::ItemStack;
+use rayon::prelude::*;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use uuid::Uuid;
+
+pub struct ItemEntity {
+    pub item_stack: ItemStack,
+    is_able_to_be_picked_up: Arc<AtomicBool>,
+    pub entity: Arc<Entity>,
+}
+
+impl ItemEntity {
+    pub async fn spawn(
+        pos: Vector3<f64>,
+        velocity: Vector3<f64>,
+        world: Arc<World>,
+        item_stack: ItemStack,
+        server: Arc<Server>,
+    ) {
+        let is_able_to_be_picked_up = Arc::new(AtomicBool::new(false));
+        {
+            let is_able_to_be_picked_up = is_able_to_be_picked_up.clone();
+            tokio::spawn(async move {
+                tokio::time::sleep(Duration::from_secs(2)).await;
+                is_able_to_be_picked_up.store(true, std::sync::atomic::Ordering::Relaxed);
+            });
+        }
+
+        let empty = Vector3 {
+            x: 0.,
+            y: 0.,
+            z: 0.,
+        };
+
+        let entity = Arc::new(Entity {
+            entity_id: server.new_entity_id(),
+            uuid: Uuid::new_v4(),
```

Comment:
There should be a check if the UUID is not matching any existing player UUID on the server, I also wonder what to do if a new player join which has the same UUID as an entity